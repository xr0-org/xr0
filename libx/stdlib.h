#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
