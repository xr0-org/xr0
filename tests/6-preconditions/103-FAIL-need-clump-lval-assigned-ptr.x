#include <stdlib.h>

int
func(int *x) ~ [ 
	pre: .clump x;
]{
	
}

int
main()
{
	int *p;
	p = 5;
	
	func(p);	/* ERROR: spec required lvalue */
}
