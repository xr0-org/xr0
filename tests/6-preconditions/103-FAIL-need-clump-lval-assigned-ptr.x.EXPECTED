6-preconditions/103-FAIL-need-clump-lval-assigned-ptr.x:16:9: cannot exec statement: parameter 'x' must be lvalue
